linked_list/LinkedList.java

```diff
@@ -0,0 +1,218 @@
+        this.next = null;
+    }
+}
+
+// Class containing methods to be performed on Linked List
+class LinkedList<T> {
+    
+    private Node<T> head;     // Pointer to first node of the Linked List
+    private int sizeOfList = 0;     // Size of the Linked List
+    
+    // Constructor to create an empty Linked List
+    public LinkedList() { 
+        head = null;
+    }
+    
+    // Method to add an element to the start of the Linked List
+    public void addFront(T input) {
+        Node<T> newNode = new Node<T>(input);     
+        newNode.next = head;    
+        head = newNode;     
+        sizeOfList++;       
+    }
+    
+    // Method to add an element to the end of the Linked List
+    public void addLast (T input) {
+        if(head == null) {     
+            Node<T> newNode = new Node<T>(input);
+            head = newNode;
+            sizeOfList = 1;
+        }
+        else {
+            Node<T> temp = head;     
+            while(temp.next != null)
+                temp = temp.next;    
+            Node<T> newNode = new Node<T>(input);
+            temp.next = newNode;    
+            sizeOfList++;
+        }
+    }
+    
+    // Method to add an element at a given index
+    public void add(T input, int index) {
+        if(index < 0 || index > sizeOfList)     
+            throw new IndexOutOfBoundsException("Invalid Index");    
+        Node<T> temp = head;
+        if(index == 0)
+            addFront(input);
+        else {
+            for (int i = 0; i < index-1; i++)  
+                temp = temp.next;
+            Node<T> newNode = new Node<T>(input);
+            newNode.next = temp.next;   
+            temp.next = newNode;    
+            sizeOfList++;
+        }
+    }
+    
+    // Method to remove the front element from the Linked List
+    public T removeFront() {
+        if(isEmpty())     
+            throw new NoSuchElementException();
+        T removedElement = head.data; 
+        head = head.next;     
+        sizeOfList--;
+        return removedElement;
+    }
+    
+    // Method to remove the last element from the Linked List
+    public T removeLast() {
+        if(isEmpty())
+            throw new NoSuchElementException();
+        T removedElement;
+        if(head.next == null) {    
+            removedElement = head.data;
+            head = null;
+        }
+        else {    
+            Node<T> temp = head;
+            while(temp.next.next != null)   
+                temp = temp.next;
+            removedElement = temp.next.data;   
+            temp.next = null;  
+        }
+        sizeOfList--;
+        return removedElement;
+    }
+    
+    // Method to remove element at specific index from the Linked List
+    public T remove(int index) {
+        if(isEmpty())
+            throw new NoSuchElementException();
+        if(index < 0 || index > sizeOfList)
+            throw new IndexOutOfBoundsException("Invalid Index");
+        if(index > 0) {
```

Comment:
You could use this as `else` part to `if` of line number 108.
Edit: Or maybe don't even add this `if` here, it wouldn't be needed anyways.